src/platform/cmx/chal_cpu.h

```diff
@@ -0,0 +1,133 @@
+#ifndef CHAL_CPU_H
+#define CHAL_CPU_H
+
+#include <pgtbl.h>
+#include <thd.h>
+#include "stm32f7xx_hal.h"
+
+extern u32_t comp1_stack[1000];
+
+/* FIXME: I doubt these flags are really the same as the PGTBL_* macros */
+static inline u32_t
+chal_cpu_fault_errcode(struct pt_regs *r) { return r->r14_lr; }
+
+static inline u32_t
+chal_cpu_fault_ip(struct pt_regs *r) { return r->r15_pc; }
+
+extern unsigned int _c1_sdata;
+extern unsigned int _c1_edata;
+extern unsigned int __c1_bss_start__;
+extern unsigned int __c1_bss_end__;
+
+static inline void //__attribute__((optimize("O3")))
+chal_user_upcall(void *ip, u16_t tid)
+{
+	long long total_swt_cycles = 0;
+	long long start_swt_cycles = 0, end_swt_cycles = 0;
+	int i;
+	char str[100];
```

Comment:
why is this here?  Remove unused variables.  You should get a warning for this; why are you not?